File: src/main/java/linq/collections/enumerable/EmptyEnumerable.java
package linq.collections.enumerable;

import linq.Enumerable;
import linq.Enumerator;
import linq.collections.enumerator.EmptyEnumerator;

public class EmptyEnumerable<T> extends Enumerable<T> {
    @Override
    public Enumerator<T> enumerator() {
        return new EmptyEnumerator<>();
    }
}

File: src/main/kotlin/linq/collections/enumerable/EmptyEnumerable.kt
package linq.collections.enumerable

import linq.Enumerable
import linq.Enumerator
import linq.collections.enumerator.EmptyEnumerator

class EmptyEnumerable<T> : Enumerable<T>() {
    override fun enumerator(): Enumerator<T> = EmptyEnumerator()
}